run-build-functions.sh

```diff
@@ -105,34 +105,50 @@ install_deps() {
 }
 
 restore_node_modules() {
-  local workspace_output
-  local workspace_exit_code
   local installer=$1
-  # YARN_IGNORE_PATH will ignore the presence of a local yarn executable (i.e. yarn 2) and default
-  # to using the global one (which, for now, is always yarn 1.x). See https://yarnpkg.com/configuration/yarnrc#ignorePath
-  # we can actually use this command for npm workspaces as well
-  workspace_output="$(YARN_IGNORE_PATH=1 "$INTERNAL_YARN_PATH/yarn" workspaces --json info 2>/dev/null)"
-  workspace_exit_code=$?
-  if [ $workspace_exit_code -eq 0 ]
-  then
-    echo "$installer workspaces detected"
-    local package_locations
-    # Extract all the packages and respective locations. .data will be a JSON object like
-    # {
-    #   "my-package-1": {
-    #     "location": "packages/blog-1",
-    #     "workspaceDependencies": [],
-    #     "mismatchedWorkspaceDependencies": []
-    #   },
-    #   (...)
-    # }
-    # We need to cache all the node_module dirs, or we'll always be installing them on each run
-    mapfile -t package_locations <<< "$(echo "$workspace_output" | jq -r '.data | fromjson | to_entries | .[].value.location')"
-    restore_js_workspaces_cache "${package_locations[@]}"
-  else
-    echo "No $installer workspaces detected"
-    restore_cwd_cache node_modules "node modules"
-  fi
+
+	if has_feature_flag "$featureFlags" "build-image_use_new_package_manager_detection"; then
+      if [ "$NETLIFY_BUILD_DEBUG" ]; then
+        echo "BUILD_INFO:\n$buildinfo\n"
+      fi
+
+			local workspaces=($(echo "$buildInfo" | jq -r '.jsWorkspaces | join(" ")'))
```

Comment:
would love that but TBH I would do this when porting this logic to buildbot. 

we had not had the jqs stderr to `/dev/null` previously so I think it won't make a difference for now. We control the serialized json that is passed to build-image through buildbot – so the possibility of parsing a invalid json is nearly going to zero

Reply:
We've had occurrences where `build-info's` execution returned messed up content for wtv reason (e.g. https://github.com/netlify/buildbot/blob/4e70b22d9ab5020b3331e75b797a5ca397c600f1/script/run-build.sh#L65) however as you've said this is different since we're the ones running it and serializing its output to JSON (instead of just using wtv it writes to stdout), so I'll defer it to you, happy to go either way 👍